### DISCUSSION TITLE
[Feature Request] Always search through all notes/tags by default

### DISCUSSION THREAD

**Discussion by chdezmar** [standardnotes/forum#2535] [Suggestions]
Global search vs Tag search would be a great addition. 
I always find myself clicking 'All notes' to be able to search globally. It is pretty annoying when you can't find what you are looking for if you have a ton of tags because you are currently in a specific Tag,

WDYT?

**Comment by atmoio** [comment]
@chdezmar I think the best way to view the current search bar is to call it "simple search bar". In the future, we might benefit from making that area extensible, so we can easily create extensions like "Advanced Search Bar". It's such a subjective area in terms of how people want and expect it to behave that making it extensible might be the best solution. But we're still quite a way from that.

**Comment by chdezmar** [comment]
@mobitar thanks for the quick reply. But maybe just a simple checkbox that would focus your search to the currently selected tag would be enough?

I can't think of any use cases for an 'Advanced Search Bar', since you can already sort by a a few options. 

btw loving this project :)

**Comment by atmoio** [comment]
I do agree that it sounds simple, but a "simple check box" here and a simple check box there and all of a sudden you find yourself swimming in unmanageable bloat. It's been a really long time since we added any new core features (Privileges perhaps being the last), so I'm just very careful in general about introducing new features and new UI, no matter the utility.

All that to say, hang in there—I'm sure we'll get this utility in one way or another ;)

**Comment by virtuallyvlad** [comment]
Would be nice to have more visual indication that you are searching only within a tag.  I've frequently come up short searching for a bunch of terms only to later realize -- d'oh! -- I had a tag clicked the whole time.

Most apps put some indication within the search bar itself or directly underneath, but do understand the concerns around anti-bloat and new UI elements.

**Comment by atmoio** [comment]
I would have thought that it saying the tag name with a search box under it would be a sufficient visual indicator that you're in that context. The Mail.app on macOS for example behaves similarly. The global search box only searches the context you're in, without really telling you that's what it's doing.

**Comment by virtuallyvlad** [comment]
Perhaps it's the grayscale tones that cause these elements to blend more into the background.  I guess I'm used to contrasting/highlighting contextual elements.
